hedera-node/hedera-app/src/main/java/com/hedera/node/app/hints/HintsLibrary.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2024-2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.hints;
+
+import com.hedera.cryptography.bls.BlsKeyPair;
+import com.hedera.cryptography.bls.BlsPrivateKey;
+import com.hedera.cryptography.bls.BlsPublicKey;
+import com.hedera.cryptography.bls.BlsSignature;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Map;
+
+/**
+ * The cryptographic operations required by the {@link HintsService}.
+ */
+public interface HintsLibrary {
+    /**
+     * Generates a new BLS key pair.
+     * @return the key pair
+     */
+    BlsKeyPair newBlsKeyPair();
+
+    /**
+     * Signs the given message with the given private key.
+     * @param message the message
+     * @param key the private key
+     * @return the signature
+     */
+    BlsSignature signPartial(@NonNull Bytes message, @NonNull BlsPrivateKey key);
+
+    /**
+     * Verifies the given signature for the given message and public key.
+     * @param message the message
+     * @param signature the signature
+     * @param publicKey the public key
+     * @return true if the signature is valid; false otherwise
+     */
+    boolean verifyPartial(@NonNull Bytes message, @NonNull BlsSignature signature, @NonNull BlsPublicKey publicKey);
+
+    /**
+     * Aggregates the signatures for the given party ids with the given aggregation key.
+     * @param aggregationKey the aggregation key
+     * @return the aggregated signature
+     */
+    Bytes aggregateSignatures(@NonNull Bytes aggregationKey, @NonNull Map<Long, BlsSignature> signatures);
```

Comment:
We need a Map here because the `GroupElement` inside `BlsSignature` is different from party id or node id ?